h5grove/content.py

```diff
@@ -1,6 +1,16 @@
 import contextlib
 from pathlib import Path
-from typing import Any, Callable, Dict, Generic, Optional, Sequence, TypeVar, Union
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Generic,
+    Optional,
+    Sequence,
+    TypeVar,
+    Union,
+    cast,
+)
```

Comment:
Most of those imported types seems to not be used.
Also, there is a lighter syntax and no need to import:
- `Dict` -> `dict`, ...
- `Union` -> `|`
- `Optional` -> `| None`

Depending on the versions of Python, you may need to add `from __future__ import annotations` as the first line of this file